### PR TITLE
Define the Horizon layer and improve the HTTP client implementation.

### PR DESCRIPTION
**Pull request by juanhurtado10**
## Context
The initial structure of this layer does not allow to define other Horizon implementations.

## Objectives
* To properly define the Horizon layer and improve the HTTP client implementation.
* To use canned elixir modules for testing.
* To remove unnecessary dependencies like `mox`.

### CODE REVIEW

**Comment by juanhurtado10** [comment]
@luishurtado this is ready for review! Thanks.

File: lib/horizon/client.ex [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-              options :: options()
-            ) :: response() | response_with_body() | response_error()
+  @impl true
+  def request(method, path, headers \\ [], body \\ "", opts \\ []),
+    do: impl().request(method, path, headers, body, opts)
+
+  @spec impl() :: atom()
+  defp impl do
+    impl = Application.get_env(:stellar_sdk, :http_client_impl)
+    if impl, do: impl, else: Client.Hackney
```
Comment (luishurtado) [inline lib/horizon/client.ex:16]: This is a shorter version:

```elixir
Application.get_env(:stellar_sdk, :http_client_impl, Client.Hackney)
```

**Review by luishurtado — CHANGES_REQUESTED** [review]
Some ideas to consider:

1. Rename `Client.Hackey` to `Client.Default`. Meaning the default implementation will use whatever HTTP Client injected via the `http_client` config variable.
2. The `Horizon.Client.request/5` forwards the invocation to the implementation, so ensure the canned function sends (`send/2`) a message to the current process (the test case, `self()`), then `assert_receive` the message in the test. If the message arrives, the Client is forwarding the call to the implementation; that is all we need to test.

**Comment by juanhurtado10** [comment]
@luishurtado Changes were implemented.

Please let me know if I'm following you on this one! 
> 2\. The `Horizon.Client.request/5` forwards the invocation to the implementation, so ensure the canned function sends (`send/2`) a message to the current process (the test case, `self()`), then `assert_receive` the message in the test. If the message arrives, the Client is forwarding the call to the implementation; that is all we need to test.

**Review by luishurtado — APPROVED** [review]

**Pull request merged**